### PR TITLE
Build: Bundle lib/docs-tools & lib/instrumenter with ts-up

### PR DESCRIPTION
**Pull request by darenbt**
Issue:

Reference: https://github.com/storybookjs/storybook/issues/18732

## What I did
Converted `lib/docs-tools` and `lib/instrumenter` to use tsup bundling script.
## How to test

- [ ] Is this testable with Jest or Chromatic screenshots? No
- [ ] Does this need a new example in the kitchen sink apps? No
- [ ] Does this need an update to the documentation? No

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

**Comment by ndelangen** [comment]
Hello @darenbt Thank you for contributing!

I am trying to update your feature branch with new changes in `next`-branch, but GitHub is refusing:
```
(build/lib-docs-tools+lib-instrumenter-tsup)⚡ % git push                                                                                                              ~/Projects/Storybook/core/code
Enumerating objects: 44, done.
Counting objects: 100% (44/44), done.
Delta compression using up to 10 threads
Compressing objects: 100% (16/16), done.
Writing objects: 100% (16/16), 2.04 KiB | 2.04 MiB/s, done.
Total 16 (delta 15), reused 0 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (15/15), completed with 14 local objects.
To https://github.com/darenbt/storybook.git
 ! [remote rejected]       build/lib-docs-tools+lib-instrumenter-tsup -> build/lib-docs-tools+lib-instrumenter-tsup (refusing to allow an OAuth App to create or update workflow `.github/workflows/generate-repros-next.yml` without `workflow` scope)
error: failed to push some refs to 'https://github.com/darenbt/storybook.git'
```

I don't think there's something you can do to make that work, but perhaps you could merge in `next` from the upstream remote yourself, to see if the recent changes in `next` fix the CI issues on this PR? 🙏 Thanks!

**Review by IanVS — APPROVED** [review]
Looks good to me!  Thanks @darenbt!

File: code/lib/instrumenter/package.json [hunk 1]
```diff
@@ -54,6 +54,7 @@
   "publishConfig": {
     "access": "public"
   },
+build/lib-docs-tools+lib-instrumenter-tsup
```
Comment (darenbt) [inline code/lib/instrumenter/package.json:57]: Here's another instance. Just needs to be deleted and the code is as you specified.

File: code/lib/docs-tools/package.json [hunk 2]
```diff
@@ -56,6 +56,7 @@
   "publishConfig": {
     "access": "public"
   },
+build/lib-docs-tools+lib-instrumenter-tsup
```
Comment (darenbt) [inline code/lib/docs-tools/package.json:59]: Don't remember adding this line in, but I know it shouldn't be there. Also not sure if that made it fail the tests on my end.  Let me know if there's anything I can do

**Review by ndelangen — APPROVED** [review]

**Pull request merged**